modules/ui/arrow-ui-data/src/test/kotlin/arrow/ui/SumTest.kt

```diff
@@ -23,49 +27,66 @@ import arrow.typeclasses.Hash
 import arrow.ui.extensions.sum.comonad.comonad
 import arrow.ui.extensions.sum.divisible.divisible
 import arrow.ui.extensions.sum.eq.eq
+import arrow.ui.extensions.sum.eqK.eqK
 import arrow.ui.extensions.sum.hash.hash
 import io.kotlintest.properties.Gen
 import io.kotlintest.shouldBe
 
 class SumTest : UnitSpec() {
-  init {
+  fun <F, G> genk(GENKF: GenK<F>, GENKG: GenK<G>) = object : GenK<SumPartialOf<F, G>> {
+    override fun <V> genK(gen: Gen<V>): Gen<Kind<SumPartialOf<F, G>, V>> = Gen.oneOf(
+      Gen.bind(GENKF.genK(gen), GENKG.genK(gen)) { a, b ->
+        Sum.left(a, b)
+      }
 
-    val cf = { x: Int -> Sum.left(Id.just(x), Id.just(x)) }
+      // question: this does not work. should we have right side too?
```

Comment:
@1Jajen1 here i have an issue with the GenK instance. the cf would just create `Sum.left`. In the GenK i tried also to create `Sum.right`. However this causes test failures.

Reply:
Honestly if that indicates an actual bug that's great. I'm on my phone so I can't take a look atm, will do in a few hours.
This pr changes the test coverage quite a bit, so I am actually hoping to find a few bugs along the way :D
Ignore them for now. If you are sure it's a bug, comment them out and add a TODO do add tickets after this pr is done.

Reply:
ok. not sure if its a but or not. will wait for your feedback. in the meantime i can leave the second part of the generator commented out.

Reply:
This is actually a bug, and I am the one that wrote the code that failed :see_no_evil: 
Just comment out the DivisibleLaws for now and add a TODO so we won't forget to open a ticket when this gets merged.